python/cudf/cudf/core/column/column.py

```diff
@@ -2098,6 +2097,15 @@ def as_column(
                 elif is_interval_dtype(dtype):
                     sr = pd.Series(arbitrary, dtype="interval")
                     data = as_column(sr, nan_as_null=nan_as_null, dtype=dtype)
+                elif (
+                    isinstance(arbitrary, list)
+                    and len(arbitrary) > 0
+                    and any(
+                        cudf.utils.dtypes.is_column_like(arb)
+                        for arb in arbitrary
+                    )
+                ):
+                    return _create_list_column_from_sequences_list(arbitrary)
```

Comment:
Two suggestions here:

* Can we move this utility into a `ListColumn` classmethod (`ListColumn.from_sequences()`)?
* Can we handle more than just lists-of-sequences? How about iterable-of-sequences?

Sorry for the runaround -- I know it's the second time I'm asking you to move the same piece of code :(